shumai/tensor/stats.ts

```diff
@@ -22,12 +22,86 @@ export function collectStats(tensors: Tensor[]) {
     }
     for (const key of Object.keys(t.stats)) {
       if (key in stats) {
+        stats[key].count += t.stats[key].count
         stats[key].time += t.stats[key].time
         stats[key].bytes += t.stats[key].bytes
+        stats[key].ops += t.stats[key].ops
       } else {
         stats[key] = t.stats[key]
       }
     }
   }
   return stats
 }
+
+export type StatsEntry = {
+  count: number
+  time: number
+  bytes: bigint
+  ops: bigint
+}
+
+export type StatsSummary = {
+  totals: StatsEntry
+  perSec: StatsEntry
+  entriesByStack: Map<string, StatsEntry>
+  entriesByOp: Map<string, StatsEntry>
+}
+
+export function getStatsSummary(tensors: Tensor[], priorStats?: StatsSummary): StatsSummary {
+  let stats: StatsSummary = priorStats
+  if (!stats) {
+    // init
+    stats = {
+      totals: { count: 0, time: 0, bytes: 0n, ops: 0n },
+      perSec: { count: 0, time: 0, bytes: 0n, ops: 0n },
+      entriesByStack: new Map(),
+      entriesByOp: new Map()
+    }
+  }
+
+  for (const t of tensors) {
+    if (!t.stats) {
+      throw new Error(`tensor has no stats ${t.ptr}:${t.shape}`)
+    }
+
+    for (const key of Object.keys(t.stats)) {
+      const op = /^(\n)?(.*)@/.exec(key)?.[2] || key
+      const entry = t.stats[key]
+      stats.totals.count += entry.count
+      stats.totals.time += entry.time
+      stats.totals.bytes += entry.bytes
+      stats.totals.ops += entry.ops
+
+      const stackEntry = stats.entriesByStack.get(key)
```

Comment:
we'll probably want to refactor the key to be a short hash (as in https://github.com/facebookresearch/shumai/blob/main/shumai/tensor/tensor.ts#L428) and carry around the actual stack string separately.  Not necessary for this PR, but just a thought

Reply:
Not sure your motive on this one. Strings are immutable and the length has no impact on the key lookup. Computing a hash is more intensive.

Reply:
oh good point re perf, I didn't realize even long strings would be interned in JS.  I'm personally biased towards readable keys (the stacks are unwieldy to print)